### DISCUSSION TITLE
It's possible to impute/enhance peak signals for scATAC-seq dataset by signac ?

### DISCUSSION THREAD

**Discussion by PeiyuShii** [stuart-lab/signac#1043] [General]
I'd like to draw a heatmap showing peak signals in differernt groups of cells. However, due to the sparsity of scATAC-seq data, it doesn't look good. It's possible to impute/enhance peak signals by smoothing signal across nearby cells in scATAC-seq dataset just like the way MAGIC(Markov Affinity-based Graph Imputation of Cells ) does in scRNA-seq ?

**Comment by timoast** [comment]
You could try running MAGIC, I haven't used it but it should also work for scATAC-seq

What I've used in the past is a simple k-nearest neighbor smoothing of the peak matrix for visualization of individual peaks in datasets that are very sparse, there's an example here: https://github.com/timoast/nanobody/blob/master/code/bmmc_dual/trajectory.R#L80-L91

**Comment by jiangpuxuan** [comment]
Hi,
I am also interested in this issue. Did `MAGIC` perform well in scATAC-seq analysis ?

**Comment by sjasws** [comment]
Dear @PeggySze @timoast @jiangpuxuan ,
Have you implemented MAGIC in signac to smooth the activity matrix?

**Reply by timoast** [reply]
no, and no plan to

**Reply by jiangpuxuan** [reply]
> Dear @PeggySze @timoast @jiangpuxuan , Have you implemented MAGIC in signac to smooth the activity matrix?

Yes. It smoothed. But I think it's meaningless to `MAGIC` the `Genescore` matrix.

**Reply by sjasws** [reply]
> > Dear @PeggySze @timoast @jiangpuxuan , Have you implemented MAGIC in signac to smooth the activity matrix?
> 
> Yes. It smoothed. But I think it's meaningless to `MAGIC` the `Genescore` matrix.

How did you achieve that? I have now obtained the harmony integrated graph through the signac process and now try to smooth the inferred gene activity from peak on top of it, just as ArchR did. Because I found that gene activity was much better embedded on the umap after ArchR imputed by MAGIC, rather than the original data, which was very helpful in determining cluster subtypes. Now that you've done it, can you share the code or process? Are you smoothing based on your own graphics?

**Reply by jiangpuxuan** [reply]
> > > Dear @PeggySze @timoast @jiangpuxuan , Have you implemented MAGIC in signac to smooth the activity matrix?
> > 
> > 
> > Yes. It smoothed. But I think it's meaningless to `MAGIC` the `Genescore` matrix.
> 
> How did you achieve that? I have now obtained the harmony integrated graph through the signac process and now try to smooth the inferred gene activity from peak on top of it, just as ArchR did. Because I found that gene activity was much better embedded on the umap after ArchR imputed by MAGIC, rather than the original data, which was very helpful in determining cluster subtypes. Now that you've done it, can you share the code or process? Are you smoothing based on your own graphics?

I use ArchR, extracting the GeneScoreMatrix and using it to make a seurat object.
```
mtx<-getMatrixFromProject(your_proj,useMatrix = "GeneScoreMatrix",)
 mtx@assays@data@listData$GeneScoreMatrix ->mtx.data
  mtx.data@Dimnames[[1]] <-mtx@elementMetadata$name
seuset.atac <- CreateSeuratObject(mtx.data)

new.ident<-your_proj$yourClusters
new.ident %>% head()
your_proj$cellNames %>% head()

names(new.ident)<-your_proj$cellNames
new.ident<-as.factor(new.ident)
new.ident %>% unique()

seuset.atac@active.ident<-new.ident

seuset.atac<-magic(seuset.atac)
```

**Reply by sjasws** [reply]
> > > > Dear @PeggySze @timoast @jiangpuxuan , Have you implemented MAGIC in signac to smooth the activity matrix?
> > > 
> > > 
> > > Yes. It smoothed. But I think it's meaningless to `MAGIC` the `Genescore` matrix.
> > 
> > 
> > How did you achieve that? I have now obtained the harmony integrated graph through the signac process and now try to smooth the inferred gene activity from peak on top of it, just as ArchR did. Because I found that gene activity was much better embedded on the umap after ArchR imputed by MAGIC, rather than the original data, which was very helpful in determining cluster subtypes. Now that you've done it, can you share the code or process? Are you smoothing based on your own graphics?
> 
> I use ArchR, extracting the GeneScoreMatrix and using it to make a seurat object.
> 
> ```
> mtx<-getMatrixFromProject(your_proj,useMatrix = "GeneScoreMatrix",)
>  mtx@assays@data@listData$GeneScoreMatrix ->mtx.data
>   mtx.data@Dimnames[[1]] <-mtx@elementMetadata$name
> seuset.atac <- CreateSeuratObject(mtx.data)
> 
> new.ident<-your_proj$yourClusters
> new.ident %>% head()
> your_proj$cellNames %>% head()
> 
> names(new.ident)<-your_proj$cellNames
> new.ident<-as.factor(new.ident)
> new.ident %>% unique()
> 
> seuset.atac@active.ident<-new.ident
> 
> seuset.atac<-magic(seuset.atac)
> ```

ArchR has its own impute feature, so you can implement it directly on ArchR. I am very inspired by you, if pass data from signac to ArchR, it seems that it can try to implement features that ArchR has but signac does not...
